### PR TITLE
EVG-18081 Sanitize html characters for resmoke lines

### PR DESCRIPTION
**Pull request by khelif96**
EVG-18081

### Description 
Some resmoke lines often contain strings that resemble html characters. This is problematic when they are run through `parseHtml` since it will try to transform the string into an html element which does not exist. This PR adds a function called `escapeHtml`  which will replace any special characters with the ansii string representation. 

Added an allowlist to `parseHtml` i only enabled the tags we would expect to see this can be updated in the future.

### Testing 
No more console errors
Unit tests

### CODE REVIEW

File: src/components/LogRow/ResmokeRow/index.tsx [hunk 1]
```diff
@@ -26,7 +27,7 @@ const ResmokeRow = forwardRef<any, BaseRowProps>((rowProps, ref) => {
       <CollapsedRow ref={ref} {...listRowProps} numCollapsed={line.length} />
     );
   }
-  const lineContent = getLine(line);
+  const lineContent = escapeHtml(getLine(line) || "");
```
Comment (minnakt) [inline src/components/LogRow/ResmokeRow/index.tsx:30]: I wonder if we should do this when processing the resmoke lines in `LogContext/state.ts`. What do you think? 🤔

File: src/utils/renderHtml/escapeHtml.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const escapeHtml = (html: string) =>
```
Comment (minnakt) [inline src/utils/renderHtml/escapeHtml.ts:1]: sorry one more thing; I noticed the highlight doesn't appear when searching for things with escaped characters, though the search picks it up correctly

Screen Shot 2022-10-19 at 10 11 53 AM

File: src/components/LogRow/ResmokeRow/index.tsx (hunk 1 above)
Comment (khelif96) [reply src/components/LogRow/ResmokeRow/index.tsx]: I moved it to baseRow instead. All the characters are technical real characters so we need to be able to search them and replacing them in `LogContext/state.ts` will not allow us to do that anymore.

File: src/utils/renderHtml/escapeHtml.ts (hunk 2 above)
Comment (khelif96) [reply src/utils/renderHtml/escapeHtml.ts:1]: Yup i ended up moving the escape to after we find the match and wrap it in `<mark>`

**Review by minnakt — APPROVED** [review]
yeehaw 🤠

**Pull request merged**